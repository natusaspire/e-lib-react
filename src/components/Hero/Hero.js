import React, { memo } from 'react';

const getStyle = imageUrl => ({ backgroundImage: `url(${imageUrl})` });

const Hero = ({ children, imageUrl = '' }) => (
  <div className="hero">
    <div
      className="hero__image"
      style={getStyle(imageUrl)}
    >
      {children}
    </div>
  </div>
);

export default memo(Hero);
