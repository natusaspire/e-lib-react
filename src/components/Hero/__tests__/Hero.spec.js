import React from 'react';
import { shallow } from 'enzyme';

import Hero from '@/components/Hero';

describe('<Hero />', () => {
  it('should render hero content with image url', () => {
    const wrapper = shallow(
      <Hero imageUrl="image.jpeg">
        <h1>Hero</h1>
      </Hero>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render hero content without image url', () => {
    const wrapper = shallow(
      <Hero>
        <h1>Hero</h1>
      </Hero>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
