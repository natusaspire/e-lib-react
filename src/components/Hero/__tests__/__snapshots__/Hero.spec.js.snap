// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> should render hero content with image url 1`] = `
<div
  className="hero"
>
  <div
    className="hero__image"
    style={
      Object {
        "backgroundImage": "url(image.jpeg)",
      }
    }
  >
    <h1>
      Hero
    </h1>
  </div>
</div>
`;

exports[`<Hero /> should render hero content without image url 1`] = `
<div
  className="hero"
>
  <div
    className="hero__image"
    style={
      Object {
        "backgroundImage": "url()",
      }
    }
  >
    <h1>
      Hero
    </h1>
  </div>
</div>
`;
